src/cli/options.rs

```diff
@@ -1710,6 +1807,90 @@ impl Options {
         }
     }
 
+    fn parse_matches_cas_aspas_add(matches: &ArgMatches) -> Result<Options, Error> {
+        let general_args = GeneralArgs::from_matches(matches)?;
+        let my_ca = Self::parse_my_ca(matches)?;
+
+        let aspa_config_str = matches.value_of("aspa").unwrap(); // required argument
+        let aspa = AspaDefinition::from_str(aspa_config_str)?;
+
+        let command = Command::CertAuth(CaCommand::AspasAddOrReplace(my_ca, aspa));
+
+        Ok(Options::make(general_args, command))
+    }
+
+    fn parse_matches_cas_aspas_remove(matches: &ArgMatches) -> Result<Options, Error> {
+        let general_args = GeneralArgs::from_matches(matches)?;
+        let my_ca = Self::parse_my_ca(matches)?;
+        let customer_str = matches.value_of("customer").unwrap();
+        let customer =
+            AspaCustomer::from_str(customer_str).map_err(|_| Error::invalid_asn(customer_str))?;
+
+        let command = Command::CertAuth(CaCommand::AspasRemove(my_ca, customer));
+
+        Ok(Options::make(general_args, command))
+    }
+
+    fn parse_matches_cas_aspas_update(matches: &ArgMatches) -> Result<Options, Error> {
+        let general_args = GeneralArgs::from_matches(matches)?;
+        let my_ca = Self::parse_my_ca(matches)?;
+
+        let mut added = vec![];
+        let mut removed = vec![];
+
+        let customer_str = matches.value_of("customer").unwrap();
+        let customer =
+            AspaCustomer::from_str(customer_str).map_err(|_| Error::invalid_asn(customer_str))?;
+
+        if let Some(add) = matches.values_of("add") {
+            for provider_str in add {
+                let provider = ProviderAs::from_str(provider_str)
+                    .map_err(|_| Error::invalid_asn(provider_str))?;
+                added.push(provider);
+            }
+        }
+
+        if let Some(remove) = matches.values_of("remove") {
+            for provider_str in remove {
+                let provider = ProviderAs::from_str(provider_str)
+                    .map_err(|_| Error::invalid_asn(provider_str))?;
+                removed.push(provider);
+            }
+        }
+
+        let update = AspaConfigurationUpdate::new(added, removed);
+        if update.is_empty() {
+            return Err(Error::general("You MUST specify at least one of --add or --remove"));
+        }
```

Comment:
Do you also need to check for the same AS being added and removed at the same time?

Reply:
good idea, added that.. a bit higher up when looping through the removals